### PR TITLE
VBase [very simple] client with Metadata fallback

### PR DESCRIPTION
**Pull request by grovina**
For now only implemented functions that will be used in Colossus transition.
Get falls back to Metadata in case of error.

To be continued: more functions for Apps transition.

### CODE REVIEW

File: vbasemetadata/client.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"github.com/vtex/go-clients/clients"
+	"github.com/vtex/go-clients/metadata"
+	"github.com/vtex/go-clients/vbase"
+)
+
+// VBase is a limited interface for using VBase with Metadata fallback
+type VBase interface {
+	GetJSON(bucket, key string, data interface{}, metadataBucket string) (string, error)
+	SaveJSON(bucket, key string, data interface{}) (string, error)
```
Comment (iagoaraujo) [inline vbasemetadata/client.go]: What do you think about add this two methods to vbase client? Something like: `GetLegacyMetadata` and `SaveLegacyMetadata`. You are saving directly on vbase, and try to read from vbase too.

File: vbasemetadata/client.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	return &client{vbaseClient, metadataClient}, nil
+}
+
+func (cl *client) GetJSON(bucket, path string, data interface{}, metadataBucket string) (string, error) {
+	etag, err := cl.vbase.GetJSON(bucket, path, data)
+	if err != nil {
+		etag, err = cl.metadata.Get(metadataBucket, path, data)
+		if err != nil {
+			etag, err = cl.vbase.SaveJSON(bucket, path, data)
```
Comment (iagoaraujo) [inline vbasemetadata/client.go]: What about `v2` suffix? Last week I understood we gonna save the legacy buckets with `v2` with go-clients. So we have to try read with `v2` suffix, and then try to read from metadata and save with the suffix.

File: vbasemetadata/client.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	return &client{vbaseClient, metadataClient}, nil
+}
+
+func (cl *client) GetJSON(bucket, path string, data interface{}, metadataBucket string) (string, error) {
+	etag, err := cl.vbase.GetJSON(bucket, path, data)
+	if err != nil {
+		etag, err = cl.metadata.Get(metadataBucket, path, data)
+		if err != nil {
```
Comment (iagoaraujo) [inline vbasemetadata/client.go]: As we discussed, just fix this to be `err == nil`.

**Comment by grovina** [comment]
[Make Colossus and Apps use VBase instead of MetaData](https://trello.com/c/FRDTOMIh/750-make-colossus-and-apps-use-vbase-instead-of-metadata)

**Pull request merged**